### PR TITLE
Simplified getWidth, getHeight, readThumbnail and getNumThumbnail met…

### PR DESCRIPTION
**Pull request by zainnetcat**
…hods

### CODE REVIEW

File: src/main/java/com/idrsolutions/JDeliImageReader.java [hunk 1]
```diff
@@ -113,14 +115,15 @@ public int getNumImages(final boolean allowSearch) {
...
     public int getWidth(final int imageIndex) throws IOException {
-        final int w;
+        int w;
         if (delegate == null) {
+            getByteArray();
             try {
-                readMetadata();
-            } catch (final Exception ex) {
-                ex.getStackTrace();
+                w = JDeli.readDimension(bytes).width;
```
Comment (markee174) [inline src/main/java/com/idrsolutions/JDeliImageReader.java:123]: I personally not a fan of code like this which can return null and catches in exception.

I would recommend
Rectangle dimensions = JDeli.readDimension(bytes);
if(dimensions != null){
w = dimensions.width;
}

File: src/main/java/com/idrsolutions/JDeliImageReader.java [hunk 2]
```diff
@@ -113,14 +115,15 @@ public int getNumImages(final boolean allowSearch) {
...
+            } catch (Exception e) {
+                e.printStackTrace();
+                w = 0;
```
Comment (markee174) [inline src/main/java/com/idrsolutions/JDeliImageReader.java:126]: See Above

**Review by markee174 — COMMENTED** [review]
I have asked to check for null, not cascade to catch method

**Comment by zainnetcat** [comment]
@markee174 I have made suggested changes and also added jpeg thumbnail support too

**Comment by markee174** [comment]
Much better

**Comment by zainnetcat** [comment]
@markee174 Made the changes involving variable assignments and Exceptions. Happy for me to merge?

**Comment by markee174** [comment]
Looks good

**Pull request merged**